scripts/release.py

```diff
@@ -429,28 +517,55 @@ def build_parser():
 
 
 def main():
+    logging.getLogger().setLevel(logging.INFO)
+
     parser = build_parser()
     args = parser.parse_args()
 
-    if args.tag.rc == 1 or (not args.tag.rc and not args.tag.patch):
-        return (
-            'The script currently only supports patch releases and release '
-            'candidates after the initial one.'
-        )
+    if args.branch:
+        branch = str(args.branch)
+    else:
+        # Discover branch from current repository
+        try:
+            branch = str(
+                    awk(
+                        git("branch", "--no-color"),
+                        "/*/ { print $2; } ")
+                ).strip()
+        except:
+            return "The current branch cannot be discovered."
 
     proceed_text = (
         'WARNING - Running this script will:\n'
         '  - add a tag to %s\n'
+        '  - update branch %s with new version number\n'
         '  - submit an issue/pull-request to %s\n'
         'If you still wish to proceed type "YES": '
-    ) % (args.repo_url, args.docs_repo_url)
+    ) % (args.repo_url, branch,  args.docs_repo_url)
 
     proceed = raw_input(proceed_text)
 
     if proceed != 'YES':
         return
 
-    rpco_repo = Repo(url=args.repo_url, cache_dir=args.cache_dir, bare=True)
+    # Get env var/cli var for github token
+    if args.github_token:
+        token = args.github_token
+    elif os.environ.get('RPC_GITHUB_TOKEN', False):
+        token = os.environ['RPC_GITHUB_TOKEN']
+    else:
+        return "Token neither found in the CLI nor in env vars"
+
+    # Get env var/cli var for version checking
+    chk_ver = os.environ.get('RPC_VERSION_CHECK', args.rpc_version_check)
```

Comment:
This would mean the environment overrides the CLI argument, this is the opposite way round to github_token.